hal/common/file.hpp

```diff
@@ -65,7 +65,7 @@ class File
 	 * 
 	 * \return  success
 	 */
-	virtual bool open(const char* path) = 0;
+	virtual bool open(const char* path, bool new_file) = 0;
```

Comment:
In fact, I can remove the parameter from the constructor and NOT from the open function. 
This parameter is used in the logging files to tell that you do not want to overwrite an existing file but rather open a new file. This feature is needed in the data_logging case but not in the flash case (as you want to overwrite the previous values of the flash.